core/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -533,4 +575,121 @@ private static void assertExceptionAsJson(Exception e, String expectedJson) thro
             return builder;
         }, expectedJson);
     }
+
+    private static Tuple<Throwable, ElasticsearchException> randomExceptions() {
+        Throwable actual;
+        ElasticsearchException expected;
+
+        int type = randomIntBetween(0, 5);
+        switch (type) {
+            case 0:
+                actual = new ClusterBlockException(singleton(DiscoverySettings.NO_MASTER_BLOCK_WRITES));
```

Comment:
oh I see the trouble that I caused you, because not all of them are actually ElasticsearchException. Good test though, much more coverage now!